### PR TITLE
feat(core): add custom font family setting

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1255 BS-1177

https://github.com/user-attachments/assets/d44359b6-b75c-4883-a57b-1f226586feec

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7924** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7924?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7878** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7878?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7879** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7879?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/general.tsx [hunk 1]
```diff
@@ -59,6 +98,149 @@ const FontFamilySettings = () => {
...
+    </Scrollable.Root>
+  );
+});
+
+Scroller.displayName = 'Scroller';
+
+const FontMenuItems = ({ onSelect }: { onSelect: (font: string) => void }) => {
+  const systemFontFamily = useService(SystemFontFamilyService).systemFontFamily;
+  useEffect(() => {
+    systemFontFamily.loadFontList();
```
Comment (EYHN) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/general.tsx]:
```suggestion
    if (systemFontFamily.fontList$.value.length === 0)
    systemFontFamily.loadFontList();
```

File: packages/frontend/core/src/modules/system-font-family/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import type { Framework } from '@toeverything/infra';
+
+import { SystemFontFamily } from './entities/system-font-family';
+import { SystemFontFamilyService } from './services/system-font-family';
+
```
Comment (EYHN) [inline packages/frontend/core/src/modules/system-font-family/index.ts:5]: `export { SystemFontFamilyService } from './services/system-font-family';`

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/general.tsx [hunk 3]
```diff
@@ -1,48 +1,87 @@
...
+  Scrollable,
   Switch,
 } from '@affine/component';
 import {
   SettingRow,
   SettingWrapper,
 } from '@affine/component/setting-components';
 import { useAppSettingHelper } from '@affine/core/hooks/affine/use-app-setting-helper';
+import type { FontData } from '@affine/core/modules/system-font-family/entities/system-font-family';
+import { SystemFontFamilyService } from '@affine/core/modules/system-font-family/services/system-font-family';
```
Comment (EYHN) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/general.tsx]:
```suggestion
import { SystemFontFamilyService } from '@affine/core/modules/system-font-family';
```

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/general.tsx [hunk 4]
```diff
@@ -1,48 +1,87 @@
...
   type RadioItem,
+  Scrollable,
   Switch,
 } from '@affine/component';
 import {
   SettingRow,
   SettingWrapper,
 } from '@affine/component/setting-components';
 import { useAppSettingHelper } from '@affine/core/hooks/affine/use-app-setting-helper';
+import type { FontData } from '@affine/core/modules/system-font-family/entities/system-font-family';
```
Comment (EYHN) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/general.tsx]:
```suggestion
import type { FontData } from '@affine/core/modules/system-font-family';
```

File: packages/frontend/core/src/modules/system-font-family/index.ts (hunk 2 above)
Comment (EYHN) [reply packages/frontend/core/src/modules/system-font-family/index.ts:5]: `export type { FontData } from './entities/system-font-family';`

**Review by EYHN — APPROVED** [review]

**Pull request merged**